internal/internal_event_handlers.go

```diff
@@ -297,9 +297,22 @@ func (wc *workflowEnvironmentImpl) UpsertSearchAttributes(attributes map[string]
 
 	upsertID := wc.GenerateSequenceID()
 	wc.decisionsHelper.upsertSearchAttributes(upsertID, attr)
+	wc.workflowInfo.SearchAttributes = mergeSearchAttributes(wc.workflowInfo.SearchAttributes, attr)
```

Comment:
I dont like the idea of overriding the search attributes one off like this.. As part of processing the event we should have a cleaner way to update `WorkflowInfo`.. I'll need to think whats the best way to handle that..

Reply:
Ok, let me know if better option

Reply:
Actually is this even the right place to have this? The user calls upsertSearchAttributes and that should cause a call to the server. The value on the client should only get updated while processing the event - not prior to the call and the server making the change.. Am I missing something?

Reply:
You are right. Changed to update in process event.